### PR TITLE
ISSUE-101 Use directly container IP for deployment validation

### PR DESCRIPTION
**Pull request by chibenwa**
Because we play the tests in a dockerized Jenkins Slave...

### CODE REVIEW

File: workflow-job [hunk 1]
```diff
@@ -274,6 +273,10 @@ def findImapPortCmd(dockerContainer) {
     return "export JAMES_PORT=`docker inspect -f '{{(index (index .NetworkSettings.Ports \"143/tcp\") 0).HostPort}}' ${dockerContainer}`"
 }
 
+def containerIp(dockerContainer) {
+    return "export JAMES_IP=`docker inspect -f '{{(index .NetworkSettings.Networks.platform_default.IPAddress)}}' ${dockerContainer}`"
```
Comment (rouazana) [inline workflow-job]: if you export it, you will redefine it each time you call it.

File: workflow-job [hunk 2]
```diff
@@ -339,7 +342,7 @@ flows["${sha1}"] = {
     stage "Build"
     node('remote') {
         ws("/home/jenkins/build/workspace/${env.BUILD_NUMBER}") {
-		def integrationArguments = "${dockerIp} \$JAMES_PORT ${sha1ToBuild}"
+		def integrationArguments = "\$JAMES_IP 143 ${sha1ToBuild}"
```
Comment (rouazana) [inline workflow-job]: here you are using the exported value. Are you sure this is the one you expect?

File: workflow-job (hunk 1 above)
Comment (rouazana) [reply workflow-job]: JAMES_IP is a strange name given the name of the function

Comment (chibenwa) [reply workflow-job]: > JAMES_IP is a strange name given the name of the function

That is what it is: James IP address...

File: workflow-job (hunk 2 above)
Comment (chibenwa) [reply workflow-job]: Well, it works. I did not intend to fix that, just to make the damn validation deployment thing work... So I did not reinvent the wheel, and use what previously existed.

File: workflow-job [hunk 3]
```diff
@@ -385,7 +379,8 @@ flows["${sha1}"] = {
 		    provisionDataForIntegrationTest("${containers.jamesCassandra}", "${jamesCliWithOptions}")
 
 		    echo "Running integration tests on Cassandra on ${buildId}"
-		    sh "${findImapPortCassandra} && docker run ${verbose} --name=${containers.integrationCassandra} --entrypoint=\"/root/integration_tests.sh\" ${useMavenCache} --volumes-from=${containers.gitPublish} ${images.jamesCompile} ${integrationArguments}"
+                    def jamesCassandraIp = containerIp("${containers.jamesCassandra}")
```
Comment (rouazana) [inline workflow-job]: indent (one space at the start)

File: workflow-job [hunk 4]
```diff
@@ -415,7 +410,8 @@ flows["${sha1}"] = {
 		    provisionDataForIntegrationTest("${containers.jamesJPA}", "${jamesCliWithOptions}")
 
 		    echo "Running integration tests on JPA on ${buildId}"
-		    sh "${findImapPortJPA} && docker run ${verbose} --name=${containers.integrationJPA} --entrypoint=\"/root/integration_tests.sh\" ${useMavenCache} --volumes-from=${containers.gitPublish} ${images.jamesCompile} ${integrationArguments}"
+                    def jamesJPAIp = containerIp("${containers.jamesJPA}")
```
Comment (rouazana) [inline workflow-job]: idem

File: workflow-job [hunk 5]
```diff
@@ -442,7 +438,8 @@ flows["${sha1}"] = {
 		    sh "docker exec ${containers.jamesSpring} ${jamesCliWithOptionsSpring} adduser imapuser@domain password"
 
 		    echo "Running integration tests on ${buildId}"
-		    sh "${findImapPortSpring} && docker run ${verbose} --name=${containers.integrationSpring} --entrypoint=\"/root/integration_tests.sh\" ${useMavenCache} --volumes-from=${containers.gitPublish} ${images.jamesCompile} ${integrationArguments}"
+                    def jamesSpringIp = containerIp("${containers.jamesSpring}")
```
Comment (rouazana) [inline workflow-job]: idem

**Review by rouazana — APPROVED** [review]

**Review by aduprat — APPROVED** [review]

**Pull request merged**